### PR TITLE
Initial commit with working component and tests

### PR DESCRIPTION
**Pull request by danielmoll**
The idea is that we have a component that wraps child components, an open and close item, and open and close callbacks to enable the toggling the display of the child content. For example, a share bar or hamburger menu.

You'll see the below code in `index.es6` this is due to the fact that `react-click-outside` wraps `component-toggle-display` and makes it impossible to test directly. I had problems even with `.renderIntoDocument()` so I should to export the component directly for testing. Please advise if there is a better way?

```
export const TestableToggleDisplay = ToggleDisplay;
export default enhanceWithClickOutside(ToggleDisplay);
```

### CODE REVIEW

**Comment by danielmoll** [comment]
Just realised I need to add the aria specific tags, but can this still be reviewed.

**Comment by danielmoll** [comment]
Added aria stuffs and updated commit log to new style.

**Comment by danielmoll** [comment]
@keithamus would you mind looking over please.

File: example.css [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+.toggle-display * {
+  line-height: 0;
+}
+
+.toggle-display__link {
+  background-color: #e3120b;
+}
+
+[data-display="open"] .toggle-display__link {
```
Comment (keithamus) [inline example.css]: This attribute may well be on another component up the tree, so it'd be good to specify which element you're hooking off of:

```css
.toggle-display [data-display="open"] .toggle-display__link {
```

File: example.css [hunk 2]
```diff
@@ -0,0 +1,36 @@
+@import '.';
+
+.toggle-display {
+  margin-left: 2rem;
+}
+
+.toggle-display * {
```
Comment (keithamus) [inline example.css]: Avoid using `*`. Better to select what you need, especially as CSS selectors are parsed RTL

**Comment by danielmoll** [comment]
@keithamus updated.

**Pull request merged**